### PR TITLE
High contrast dark-theme

### PR DESCRIPTION
**Pull request by dem4ron**
Screenshot 2023-07-18 at 15 05 39

### CODE REVIEW

File: app/javascript/hooks/use-theme-observer.tsx [hunk 1]
```diff
@@ -63,11 +78,31 @@ export function useThemeObserver(): ThemeData {
         }))
       }
     }
-    mediaQueryList.addEventListener('change', mediaQueryListener)
+
+    const mediaQueryListAccessibility = window.matchMedia(
+      '(prefers-contrast: more)'
+    )
+    const mediaQueryListenerAccessibility = () => {
```
Comment (dem4ron) [inline app/javascript/hooks/use-theme-observer.tsx]:
```suggestion
    const mediaQueryListenerAccessibilityDark = () => {
```
(to be more explicit)

File: app/javascript/hooks/use-theme-observer.tsx [hunk 2]
```diff
@@ -63,13 +99,34 @@ export function useThemeObserver(): ThemeData {
...
-    mediaQueryList.addEventListener('change', mediaQueryListener)
+    const mediaQueryListAccessibilityDark = window.matchMedia(
+      '(prefers-contrast: more)'
+    )
+    const mediaQueryListenerAccessibilityDark = () => {
+      if (mediaQueryListAccessibilityDark.matches) {
+        replaceThemeWith('theme-accessibility-dark')
+        patchThemeDebounced('accessibility-dark', updateEndpoint)
+      } else {
+        replaceThemeWith('theme-light')
```
Comment (dem4ron) [inline app/javascript/hooks/use-theme-observer.tsx]: if the current theme is not `theme-accessibility-dark` don't replace it.

File: app/javascript/hooks/use-theme-observer.tsx [hunk 3]
```diff
@@ -63,13 +99,34 @@ export function useThemeObserver(): ThemeData {
         }))
       }
     }
-    mediaQueryList.addEventListener('change', mediaQueryListener)
+    const mediaQueryListAccessibilityDark = window.matchMedia(
```
Comment (dem4ron) [inline app/javascript/hooks/use-theme-observer.tsx]:
```suggestion
    const mediaQueryPrefersContrastMore = window.matchMedia(
```
...AccessibilityDark is a fantasy name, won't be clear enough on the long run

File: app/javascript/hooks/use-theme-observer.tsx [hunk 4]
```diff
@@ -52,9 +88,9 @@ export function useThemeObserver(): ThemeData {
 
     observer.observe(document.body, { attributes: true })
 
-    const mediaQueryList = window.matchMedia('(prefers-color-scheme: dark)')
-    const mediaQueryListener = () => {
-      const isDark = mediaQueryList.matches
+    const mediaQueryListDark = window.matchMedia('(prefers-color-scheme: dark)')
```
Comment (dem4ron) [inline app/javascript/hooks/use-theme-observer.tsx]:
```suggestion
    const mediaQueryPrefersColorSchemeDark = window.matchMedia('(prefers-color-scheme: dark)')
```
same as below

File: app/css/ui-kit/colors.css [hunk 5]
```diff
@@ -715,3 +702,197 @@ body.namespace-.controller-premium.theme-light,
...
+        }
+    }
+
+    --c-prominent-link-color: #f0f;
+    --c-prominent-link-icon-filter: invert(18%) sepia(100%) saturate(2923%)
+        hue-rotate(293deg) brightness(120%) contrast(134%);
+    --box-shadow-sm: 0 0 0 2px #000;
+    --box-shadow-base: 0 0 0 2px #000;
+    --box-shadow-lg: 0 0 0 2px #000;
+    --box-shadow-lgZ1: 0 0 0 2px #000;
```
Comment (iHiD) [inline app/css/ui-kit/colors.css:738]: That's such an obvious/nice fix :)

**Review by iHiD — APPROVED** [review]

**Pull request merged**